packages/pouchdb-core/src/setup.js

```diff
@@ -45,6 +45,8 @@ PouchDB.adapter = function (id, obj, addToPreferredAdapters) {
 PouchDB.plugin = function (obj) {
   if (typeof obj === 'function') { // function style for plugins
     obj(PouchDB);
+  } else if (Object.keys(obj).length === 0){
```

Comment:
I think you'll also want to check whether obj is an object here. e.g. `Object.keys("foo")` will return `["0", "1", "2"]` but `"foo"` is not a valid plugin. I'd suggest changing this to `} else if (typeof obj !== 'object' || Object.keys(obj).length === 0){`